dtschema/schemas/chosen.yaml

```diff
@@ -273,6 +273,28 @@ properties:
       This property is used by U-Boot to pass its version down to the operating
       system.
 
+  devicetree-path:
+    $ref: types.yaml#/definitions/string-array
+    items:
+      - $ref: types.yaml#/definitions/string
+        pattern: "^.*\\.dtb$"
+      - $ref: types.yaml#/definitions/string
+        pattern: "^.*\\.dtbo$"
+    additionalItems: true
+    minItems: 1
+    description:
+      This property describes the path to this devicetree blob relative to the
+      install directory. It may be used by bootloaders to load an alternative
+      version of the devicetree blob (e.g. the one associated with a specific
+      kernel version).
+
+      Additional items describe the path to any device tree overlays that should
+      be applied to the devicetree blob. An example is provided below.
+
+      devicetree-path = "qcom/qcs6490-rb3gen2.dtb",
+                        "qcom/qcs6490-rb3gen2-overlay1.dtbo",
+                        "qcom/qcs6490-rb3gen2-overlay2.dtbo";
```

Comment:
@kcxt
You only mention U-Boot but the property name does not point to U-Boot.

In GRUB we have a device-tree command which could load a completely different device-tree. Do you expect GRUB to update the property?

Your example uses a vendor/board.dtb form. Should absolute paths be allowable? E.g. GRUB would not know if the path contains a vendor directory. It would be easiest to just copy the argument of the devicetree command.

Reply:
Linux does not provide dtbo files. I have no clue how a loader like GRUB would find "qcom/qcs6490-rb3gen2-overlay2.dtbo" to apply the overlay.